core/trino-main/src/main/java/io/trino/type/LikePatternType.java

```diff
@@ -71,7 +72,7 @@ public Object getObject(Block block, int position)
             escape = Optional.of((char) block.getInt(position, offset));
         }
 
-        return LikeMatcher.compile(pattern, escape);
+        return DfaLikeMatcher.compile(pattern, escape);
```

Comment:
I think you have to support both matchers here because this is used to serialize function call via `io.trino.sql.planner.LiteralEncoder#toExpression`